src/main/java/com/timgroup/statsd/Telemetry.java

```diff
@@ -114,12 +158,121 @@ public void flush() {
         processor.send(new TelemetryMessage(this.packetsDroppedMetric, this.packetsDropped.getAndSet(0), tags));
         processor.send(new TelemetryMessage(this.packetsDroppedQueueMetric, this.packetsDroppedQueue.getAndSet(0), tags));
         processor.send(new TelemetryMessage(this.aggregatedContextsMetric, this.aggregatedContexts.getAndSet(0), tags));
+
+        if (devMode) {
+            processor.send(new TelemetryMessage(this.metricsByTypeSentMetric, this.gaugeSent.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.GAUGE)));
+            processor.send(new TelemetryMessage(this.metricsByTypeSentMetric, this.countSent.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.COUNT)));
+            processor.send(new TelemetryMessage(this.metricsByTypeSentMetric, this.setSent.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.SET)));
+            processor.send(new TelemetryMessage(this.metricsByTypeSentMetric, this.histogramSent.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.HISTOGRAM)));
+            processor.send(new TelemetryMessage(this.metricsByTypeSentMetric, this.distributionSent.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.DISTRIBUTION)));
+
+            processor.send(new TelemetryMessage(this.aggregatedContextsByTypeMetric, this.aggregatedGaugeContexts.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.GAUGE)));
+            processor.send(new TelemetryMessage(this.aggregatedContextsByTypeMetric, this.aggregatedCountContexts.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.COUNT)));
+            processor.send(new TelemetryMessage(this.aggregatedContextsByTypeMetric, this.aggregatedSetContexts.getAndSet(0),
+                        getTelemetryTags(tags, Message.Type.SET)));
+        }
+    }
+
+    protected String getTelemetryTags(String tags, Message.Type type) {
```

Comment:
Could we compute the tags at init time ? They should always be the same, no ?

Reply:
I mean, we could but this is called only on flushes so every X seconds, not sure caching is worth it.

Reply:
indeed :+1: